tests/sentry/integrations/slack/test_client.py

```diff
@@ -0,0 +1,128 @@
+import re
+
+import responses
+from django.test import override_settings
+from responses import matchers
+
+from sentry.integrations.slack.client import SlackClient
+from sentry.models.integrations.organization_integration import OrganizationIntegration
+from sentry.silo.base import SiloMode
+from sentry.silo.util import PROXY_BASE_PATH
+from sentry.testutils import TestCase
+
+control_address = "http://controlserver"
+secret = "hush-hush-im-invisible"
+
+
+@override_settings(
+    SENTRY_SUBNET_SECRET=secret,
+    SENTRY_CONTROL_ADDRESS=control_address,
+)
+class SlackClientTest(TestCase):
+    def setUp(self):
+        self.user_access_token = "xoxp-user-access-token"
+        self.access_token = "xoxb-access-token"
+        self.integration = self.create_integration(
+            organization=self.organization,
+            external_id="slack:1",
+            provider="slack",
+            metadata={"access_token": self.access_token},
+        )
+        self.organization_integration = OrganizationIntegration.objects.get(
+            integration_id=self.integration.id
+        )
+        self.payload = {"channel": "#announcements", "message": "i'm ooo next week"}
+        self.mock_user_access_token_response = {"ok": True, "auth": "user"}
+        self.mock_access_token_response = {"ok": True, "auth": "token"}
+        self.mock_not_authed_response = {"ok": True, "auth": None}
+        base_response_kwargs = {
+            "method": responses.POST,
+            "url": re.compile(r"\S+chat.postMessage$"),
+            "status": 200,
+            "content_type": "application/json",
+        }
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_user_access_token_response,
+            match=[
+                matchers.header_matcher(
+                    {"Authorization": f"Bearer {self.user_access_token}"},
+                )
+            ],
+        )
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_access_token_response,
+            match=[matchers.header_matcher({"Authorization": f"Bearer {self.access_token}"})],
+        )
+        responses.add(
+            **base_response_kwargs,
+            json=self.mock_not_authed_response,
+            match=[matchers.header_matcher({})],
+        )
+
+    @responses.activate
+    def test_integration_proxy_is_active(self):
+        class SlackProxyTestClient(SlackClient):
+            use_proxy_url_for_tests = True
+
```

Comment:
This attribute needs to be set prior to running `__init__` to ensure requests get proxied for tests. This allows us to test against the proxy url instead of `https://slack.com/api`